packages/datatrak-web/src/features/Survey/SurveyContext.tsx

```diff
@@ -21,20 +21,66 @@ const convertNumberToLetter = (number: number) => {
 };
 
 type SurveyFormContextType = {
-  formData: { [key: string]: any };
-  setFormData: (data: { [key: string]: any }) => void;
+  formData: Record<string, any>;
   activeScreen: SurveyScreenComponent[];
   isLast: boolean;
   numberOfScreens: number;
   screenNumber: number;
   screenHeader?: string;
   displayQuestions: SurveyScreenComponent[];
+  sideMenuOpen?: boolean;
+  surveyScreenComponents?: Record<number, SurveyScreenComponent[]>;
 };
 
-const SurveyFormContext = createContext({} as SurveyFormContextType);
+const defaultContext = {
+  formData: {},
+  activeScreen: [],
+  isLast: false,
+  numberOfScreens: 0,
+  screenNumber: 1,
+  screenHeader: '',
+  displayQuestions: [],
+  sideMenuOpen: false,
+  surveyScreenComponents: [],
+} as SurveyFormContextType;
+
+const SurveyFormContext = createContext(defaultContext as SurveyFormContextType);
+
+export enum ACTION_TYPES {
+  SET_FORM_DATA = 'SET_FORM_DATA',
+  TOGGLE_SIDE_MENU = 'TOGGLE_SIDE_MENU',
+}
+
+interface SurveyFormAction {
+  type: ACTION_TYPES;
+  payload?: Record<string, any> | null;
+}
+
+export const SurveyFormDispatchContext = createContext<Dispatch<SurveyFormAction> | null>(null);
+
+export const surveyReducer = (
+  state: SurveyFormContextType,
+  action: SurveyFormAction,
+): SurveyFormContextType => {
+  switch (action.type) {
+    case ACTION_TYPES.SET_FORM_DATA:
+      return {
+        ...state,
+        formData: action.payload as Record<string, any>,
+      };
```

Comment:
```suggestion
      return {
        ...state,
        formData: { ...state.formData, action.payload as Record<string, any> },
      };
```
What do you think about doing it like this so that when we call `setFormData` we just pass in the new form data and don't have to re-create the whole formData object everytime?

Reply:
Sure, can do

Reply:
I actually initially planned to make it pass in a key and a value, but I decided against it for the occasions where we want to reset multiple things

Reply:
Yes good decision I think we will be passing in a screen of values at a time

Reply:
ps. this isn't really related but I noticed while testing that on the radio buttons, clicking on the text doesn't trigger the onClick event? If it's a quick fix, would you mind fixing that up to?